pkg/cmd/cli/cmd/env.go

```diff
@@ -268,29 +285,50 @@ func RunEnv(f *clientcmd.Factory, in io.Reader, out io.Writer, cmd *cobra.Comman
 		}
 	}
 	if one && skipped == len(infos) {
-		return fmt.Errorf("the %s %s is not a pod or does not have a pod template", infos[0].Mapping.Resource, infos[0].Name)
+		return fmt.Errorf("%s/%s is not a pod or does not have a pod template", infos[0].Mapping.Resource, infos[0].Name)
 	}
 
 	if list {
 		return nil
 	}
 
-	objects, err := resource.AsVersionedObject(infos, false, outputVersion)
-	if err != nil {
-		return err
-	}
-
 	if len(outputFormat) != 0 {
+		objects, err := resource.AsVersionedObjects(infos, outputVersion)
+		if err != nil {
+			return err
+		}
 		p, _, err := kubectl.GetPrinter(outputFormat, "")
 		if err != nil {
 			return err
 		}
-		return p.PrintObj(objects, out)
+		for _, object := range objects {
+			if err := p.PrintObj(object, out); err != nil {
+				return err
+			}
+		}
+		return nil
+	}
+
+	apiVersion := cmdutil.GetFlagString(cmd, "api-version")
```

Comment:
`--api-version` modifies clientConfig.Version, I don't think we need to check this directly